plugins/generator-1.20.1/forge-1.20.1/templates/modbase/accesstransformer.cfg.ftl

```diff
@@ -12,10 +12,13 @@ public net.minecraft.world.level.levelgen.feature.ScatteredOreFeature <init>(Lco
 public-f net.minecraft.world.level.levelgen.feature.TreeFeature m_142674_(Lnet/minecraft/world/level/levelgen/feature/FeaturePlaceContext;)Z #place
 </#if>
 
-<#if w.hasElementsOfType("livingentity")>
+<#if w.getGElementsOfType("livingentity")?filter(e -> e.canTrade)?size != 0>
 public-f net.minecraft.world.entity.npc.Villager f_35366_ #assignProfessionWhenSpawned
 public-f net.minecraft.world.entity.npc.Villager f_35373_ #updateMerchantTimer
 public-f net.minecraft.world.entity.npc.Villager f_35374_ #increaseProfessionLevelOnUpdate
 public net.minecraft.world.entity.npc.Villager m_35528_()V #increaseMerchantCareer
 public-f net.minecraft.world.entity.npc.Villager f_35375_ #lastTradedPlayer
+<#if w.getGElementsOfType("livingentity")?filter(e -> e.villagerTradingType)?size != 0>
```

Comment:
Given all other transformers are applied to the same class, maybe the two conditions should be unified?
`e -> e.canTrade && e.villagerTradingType`

Reply:
Done!